planner/core/planbuilder.go

```diff
@@ -3283,6 +3283,16 @@ func (b *PlanBuilder) buildSimple(ctx context.Context, node ast.StmtNode) (Plan,
 			p.StaleTxnStartTS = readTS
 			// consume read ts here
 			b.ctx.GetSessionVars().TxnReadTS.UseTxnReadTS()
+		} else if b.ctx.GetSessionVars().EnableExternalTSRead {
```

Comment:
should check `EnableExternalTSRead` here ?

Reply:
Nice catch! Added a check for `!b.ctx.GetSessionVars().InRestrictedSQL`